src/wazuh_db/wdb_global.c

```diff
@@ -1418,9 +1446,12 @@ int wdb_global_recalculate_agent_groups_hash_without_sync_status(wdb_t* wdb, int
         mdebug1("No groups in belongs table for agent '%03d'", agent_id);
     }
 
-    if (WDBC_ERROR == wdb_global_set_agent_group_hash(wdb, agent_id, agent_groups_csv, agent_groups_csv ? groups_hash : NULL)) {
-        result = WDBC_ERROR;
-        merror("There was an error assigning the groups hash to agent '%03d'", agent_id);
+    // if the previous group is different from the new one, we update the agent group context
```

Comment:
Another improvement that can be considered is avoiding the hash calculation in case the groups haven't changed.

Reply:
Yeap, you mean not only check if the wdb_global_calculate_agent_group_csv return a non-null value.

Reply:
We can leave it for another moment.
We don't have time.